File: backend/src/main/java/ma/investpro/repository/CommissionRepository.java
package ma.investpro.repository;

import ma.investpro.entity.Commission;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

@Repository
public interface CommissionRepository extends JpaRepository<Commission, Long> {

    Optional<Commission> findByDepenseId(Long depenseId);

    List<Commission> findByConventionId(Long conventionId);

    List<Commission> findByDateCalculBetween(LocalDate dateDebut, LocalDate dateFin);

    @Query("SELECT c FROM Commission c WHERE " +
           "c.depense.projet.id = :projetId")
    List<Commission> findByProjetId(@Param("projetId") Long projetId);

    @Query("SELECT c FROM Commission c WHERE " +
           "c.depense.fournisseur.id = :fournisseurId")
    List<Commission> findByFournisseurId(@Param("fournisseurId") Long fournisseurId);

    @Query("SELECT SUM(c.montantCommissionTtc) FROM Commission c")
    Double sumMontantCommissionTtc();

    @Query("SELECT SUM(c.montantCommissionTtc) FROM Commission c WHERE " +
           "c.dateCalcul BETWEEN :dateDebut AND :dateFin")
    Double sumMontantCommissionTtcByPeriode(
        @Param("dateDebut") LocalDate dateDebut,
        @Param("dateFin") LocalDate dateFin
    );
}

File: backend/src/main/kotlin/ma/investpro/repository/CommissionRepository.kt
package ma.investpro.repository

import ma.investpro.entity.Commission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface CommissionRepository : JpaRepository<Commission, Long> {

    /**
     * Recherche par dépense
     */
    fun findByDepenseId(depenseId: Long): Commission?

    /**
     * Recherche par convention
     */
    fun findByConventionId(conventionId: Long): List<Commission>

    /**
     * Recherche par plage de dates
     */
    @Query("SELECT c FROM Commission c WHERE c.dateCalcul BETWEEN :dateDebut AND :dateFin ORDER BY c.dateCalcul DESC")
    fun findByDateCalculBetween(
        @Param("dateDebut") dateDebut: LocalDate,
        @Param("dateFin") dateFin: LocalDate
    ): List<Commission>

    /**
     * Recherche multi-critères
     */
    @Query("""
        SELECT c FROM Commission c
        WHERE (:conventionId IS NULL OR c.convention.id = :conventionId)
        AND (:dateDebut IS NULL OR c.dateCalcul >= :dateDebut)
        AND (:dateFin IS NULL OR c.dateCalcul <= :dateFin)
        ORDER BY c.dateCalcul DESC
    """)
    fun search(
        @Param("conventionId") conventionId: Long?,
        @Param("dateDebut") dateDebut: LocalDate?,
        @Param("dateFin") dateFin: LocalDate?
    ): List<Commission>

    /**
     * Total des commissions par convention
     */
    @Query("""
        SELECT COALESCE(SUM(c.montantCommissionTtc), 0)
        FROM Commission c
        WHERE c.convention.id = :conventionId
    """)
    fun getTotalByConvention(@Param("conventionId") conventionId: Long): java.math.BigDecimal
}
